src/pages/filter/ui/modal-edit/components/modal-filters/modal-filters.tsx

```diff
@@ -0,0 +1,127 @@
+import { ReactElement, useEffect } from 'react'
+import { toJS } from 'mobx'
+import { observer } from 'mobx-react-lite'
+
+import { ActionType } from '@declarations'
+import { t } from '@i18n'
+import dtreeStore from '@store/dtree'
+import { Pagintaion } from '@components/pagintaion'
+import activeStepStore from '@pages/filter/active-step.store'
+import { SelectModalButtons } from '@pages/filter/ui/query-builder/ui/select-modal-buttons'
+import { createChunks } from '@utils/createChunks'
+import dtreeModalStore from '../../../../modals.store'
+import { QueryBuilderSearch } from '../../../query-builder/query-builder-search'
+import { HeaderModal } from '../../../query-builder/ui/header-modal'
+import { ModalBase } from '../../../query-builder/ui/modal-base'
+import modalEditStore from '../../modal-edit.store'
+import { EditModalButtons } from '../edit-modal-buttons'
+import { FiltersList } from './components/filter-list'
+import { FiltersMods } from './components/filter-mods'
+import modalFiltersStore from './modal-filters.store'
+
+export const ModalFilters = observer((): ReactElement => {
+  const { groupName, currentStepGroups } = modalEditStore
+
+  const currentStepIndex = activeStepStore.activeStepIndex
+  const currentGroupIndex = dtreeModalStore.groupIndexToChange
+
+  const currentGroup =
+    dtreeStore.stepData[currentStepIndex].groups[currentGroupIndex]
+
+  const currentGroupToModify = dtreeStore.stepData[currentStepIndex].groups
+
+  const { searchValue, currentPage, groupsPerPage } = modalFiltersStore
+
+  const originGroupList: [string, number][] =
+    toJS(dtreeStore.selectedGroups[2]) ?? []
+
+  const filteredGroupList = originGroupList.filter(
+    (variant: [string, number]) =>
+      variant[0]
+        .toLocaleLowerCase()
+        .includes(modalFiltersStore.searchValue.toLocaleLowerCase()),
+  )
+
+  const chunks = createChunks(filteredGroupList, groupsPerPage)
```

Comment:
Slicing array into chunks every time is very bad for performance.  And also, `createChunks` function is terrible, because it  contains this code: `JSON.parse(JSON.stringify(items))`).

Please see how it is implemented in `panels/enum-panel.tsx`